### PR TITLE
Remove and forbid use of com.google.common.net.InetAddresses

### PR DESCRIPTION
**Pull request by wittyameta**
This commit removes and now forbids all uses of com.google.common.net.InetAddresses across the codebase. This is one of many steps in the eventual removal of Guava as a dependency.

Relates #13224

### CODE REVIEW

**Comment by nik9000** [comment]
The automatic CLA check is failing here because at least some of these commits are under a different email address than the one that has signed the CLA. That is fine - I checked against the list of CLA signers and @wittyameta is on it.

File: core/pom.xml [hunk 1]
```diff
@@ -194,6 +194,11 @@
             <artifactId>jna</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
```
Comment (nik9000) [inline core/pom.xml]: I think we've been trying to remove dependencies so adding this to core in exchange for some Guava classes is probably going to upset someone.

Comment (wittyameta) [inline core/pom.xml]: I thought this was specifically for Guava dependencies. Are we going to replace every Guava functionality by our own implementation?

File: core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java [hunk 2]
```diff
@@ -180,4 +182,13 @@ static String format(InetAddress address, int port, boolean includeHost) {
...
         return builder.toString();
     }
+    
+    @SuppressForbidden(reason = "we call toString to avoid a DNS lookup")
+    private static String getInet6HostAddress(InetAddress address) {
+        return address.getHostAddress().replaceAll(":0+([0-9])",":$1")
+                .replaceAll("((?::0\\b){2,}):?(?!\\S*\\b\\1:0\\b)(\\S*)", "::$2")
+                .replaceFirst("^0::", "::")
+                .replaceFirst("%\\d+", "");
+        // see http://stackoverflow.com/a/7044170/4746692
```
Comment (rmuir) [inline core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java]: I don't think its good to implement address formatting with some regex from stackoverflow.

we need solid code that follows the RFCs: which is what the guava code does. We should not have to get "incorrect" formatting of internet addresses just to remove guava. We will keep guava around as long as we can until things are correct.

Comment (wittyameta) [inline core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java]: Yeah, I agree with the issue with regex formatting.

Regarding incorrect formatting, the tests did look like they covered all the scenarios.

File: core/pom.xml (hunk 1 above)
Comment (jasontedor) [inline core/pom.xml]: Our goal is to minimize dependencies, and especially commonly used non-critical dependencies. I wouldn't go so far as to say adding a new dependency as part of removing Guava will upset someone, but @nik9000's tacit point that this here defeats the purpose is correct.

**Pull request closed without merging**